plugins/modules/alert_channel.py

```diff
@@ -0,0 +1,710 @@
+            event_channel_id = channel_details['name']
+            if not self.module.check_mode:
+                self.event_api.create_event_channel_0(event_channel, event_channel_id=event_channel_id)
+            return True
+        except Exception as e:
+            msg = f"Failed to send test alert for alert channel {event_channel_id} with error: {utils.determine_error(e)}"
+            LOG.error(msg)
+            self.module.fail_json(msg=msg)
+
+    def get_alert_channel_details(self, name):
+        """Get alert channel details
+        :param name: Name of the alert channel
+        :return: Dictionary of alert channel details
+        """
+        LOG.info("Getting alert channel details.")
+        channel_details = Events(self.event_api, self.module).get_alert_channel(alert_channel_id=name)
+        if channel_details:
+            return channel_details['channels'][0]
+        return None
+
+    def delete_alert_channel(self, channel_details):
+        """
+        Delete the alert channel.
+        :param channel_details: Dictionary of alert channel details
+        :return: True if the operation is successful.
+        """
+        try:
+            LOG.info("Deleting alert channel.")
+            event_channel_id = channel_details['name']
+            if not self.module.check_mode:
+                self.event_api.delete_event_channel(event_channel_id=event_channel_id)
+            return True
+
+        except Exception as e:
+            msg = f"Failed to delete alert channel {event_channel_id} with error: {utils.determine_error(e)}"
+            LOG.error(msg)
+            self.module.fail_json(msg=msg)
+
+    def get_diff_after(self, channel_params, channel_details):
+        """Get diff between playbook input and alert channel details
+        :param channel_params: Dictionary of parameters input from playbook
+        :param channel_details: Dictionary of alert channel details
+        :return: Dictionary of parameters of differences"""
+
+        if channel_params["state"] == "absent":
+            return {}
+        else:
+            if channel_details is None:
+                diff_dict = {
+                    "allowed_nodes": channel_params['allowed_nodes'],
+                    "enabled": channel_params['enabled'],
+                    "excluded_nodes": channel_params['excluded_nodes'],
+                    "id": "",
+                    "name": channel_params['name'],
+                    "parameters": channel_params['smtp_parameters'],
+                    "system": False,
+                    "type": channel_params['type']
+                }
+                return diff_dict
+            else:
+                diff_dict = copy.deepcopy(channel_details)
+                modify_dict = self.is_alert_channel_modify_required(channel_params, diff_dict)
+                for key in modify_dict.keys():
+                    diff_dict[key] = modify_dict[key]
+                return diff_dict
```

Comment:
As at line 504 , we return the same diff_dict , lets keep return statement outside of if else statements.

Reply:
Fixed.